library/src-3.x/scala/internal/quoted/Matcher.scala

```diff
@@ -83,6 +83,14 @@ object Matcher {
       /** Normalieze the tree */
       def normalize(tree: Tree): Tree = tree match {
         case Block(Nil, expr) => normalize(expr)
+        case Block(stats, expr) if stats.exists(IsTypeDef.unapply(_).isDefined) =>
+          // TODO check that these definitions are quote taging artifacts
```

Comment:
Need to rebase on to of #6700 to get the [`@quoteTypeTag`](https://github.com/lampepfl/dotty/pull/6700/files#diff-7b199cbe62cc044a10a26459c0f0fa46R119) to be able to check this.